src/util.ts

```diff
@@ -769,19 +770,81 @@ export function validateMinimumIncrease(proposed: string, min: string) {
 }
 
 /**
- * Extracts content identifier from ipfs url.
+ * Removes ipfs protocol prefix from ipfs-url.
  *
- * @param url - Ipfs url.
- * @returns Ipfs content identifier as string.
+ * @param ipfsUrl - ipfs url
+ * @returns Ipfs content identifier and (possibly) path in a string
+ * @throws will throw if the url passed is not ipfs.
  */
-export function getIpfsUrlContentIdentifier(url: string): string {
-  if (url.startsWith('ipfs://ipfs/')) {
-    return url.replace('ipfs://ipfs/', '');
-  }
+export function removeIpfsProtocolPrefix(ipfsUrl: string) {
+  if (ipfsUrl.startsWith('ipfs://ipfs/')) {
+    return ipfsUrl.replace('ipfs://ipfs/', '');
+  } else if (ipfsUrl.startsWith('ipfs://')) {
+    return ipfsUrl.replace('ipfs://', '');
+  }
+  // this method should not be used with non-ipfs urls (i.e. startsWith('ipfs://') === true)
+  throw new Error('this method should not be used with non ipfs urls');
+}
 
-  if (url.startsWith('ipfs://')) {
-    return url.replace('ipfs://', '');
+/**
+ * Extracts content identifier and path from ipfs url.
+ *
+ * @param ipfsUrl - ipfs url
+ * @returns Ipfs content identifier as string and path as string.
+ * @throws Will throw if the url passed is not ipfs.
+ */
+export function getIpfsCIDv1AndPath(
+  ipfsUrl: string,
+): { cid: string; path?: string } {
+  const url = removeIpfsProtocolPrefix(ipfsUrl);
+
+  // check if there is a path
+  // (CID is everything preceding first forward slash, path is everything after)
+  const index = url.indexOf('/');
+  const cid = index !== -1 ? url.substring(0, index) : url;
+  const path = index !== -1 ? url.substring(index) : '';
+
+  // we want to ensure that the CID is v1 (https://docs.ipfs.io/concepts/content-addressing/#identifier-formats)
+  return {
+    cid: CID.parse(cid).toV1().toString(),
+    path,
+  };
+}
+
+/**
+ * Adds URL protocol prefix to url string if missing.
+ *
+ * @param urlString - Ipfs url.
+ * @returns string.
+ */
+export function addUrlProtocolPrefix(urlString: string): string {
+  if (!urlString.match(/(^http:\/\/)|(^https:\/\/)/u)) {
+    return `https://${urlString}`;
   }
+  return urlString;
+}
 
-  return url;
+/**
+ * Formats url correctly for use retrieving assets hosted on IPFS.
+ *
+ * @param ipfsGateway - the user preferred ipfsGateway.
```

Comment:
Nit: These descriptions could be better. Keep in mind that these aren't just code comments, they are inline documentation. This gets used to build our API documentation (which isn't automatically built/hosted right now, but it will be soon, and it can be generated manually). Better to use proper case (start the sentence with a capital letter, capitalize acronyms like IPFS, use words and not variable names, etc.), and to include relevant details like what sort of string this is.

For example, it looks like this is expected to be the URL for the IPFS gateway, except that the protocol is optional, and the trailing `/ipfs` path is optional.

The same general point applies to most of the other inline docs as well.